bus-mapping/src/rpc.rs

```diff
@@ -152,53 +189,71 @@ mod rpc_tests {
     // have a Geth client synced with mainnet to have unified "test-vectors".
     #[ignore]
     #[tokio::test]
-    async fn test_get_block_by_hash() {
-        let transport = Http::new(Url::parse("http://localhost:8545").unwrap());
-
-        let hash = Hash::from_str("0xe4f7aa19a76fcf31a6adff3b400300849e39dd84076765fb3af09d05ee9d787a").unwrap();
-        let prov = GethClient::new(transport);
-        let block_by_hash = prov.get_block_by_hash(hash).await.unwrap();
-        assert!(hash == block_by_hash.hash.unwrap());
+    async fn test_get_block_by_number() {
+        let prov = get_provider();
+        let block_by_num_latest =
+            prov.get_block_by_number(BlockNumber::Latest).await.unwrap();
+        let block_by_num = prov.get_block_by_number(2u64.into()).await.unwrap();
+        assert!(
+            block_by_num.transactions[0].hash
+                == block_by_num_latest.transactions[0].hash
+        );
     }
 
     // The test is ignored as the values used depend on the Geth instance used
     // each time you run the tests. And we can't assume that everyone will
     // have a Geth client synced with mainnet to have unified "test-vectors".
     #[ignore]
     #[tokio::test]
-    async fn test_get_block_by_number() {
-        let transport = Http::new(Url::parse("http://localhost:8545").unwrap());
-
-        let hash = Hash::from_str("0xe4f7aa19a76fcf31a6adff3b400300849e39dd84076765fb3af09d05ee9d787a").unwrap();
-        let prov = GethClient::new(transport);
+    async fn test_get_block_by_hash() {
+        let prov = get_provider();
         let block_by_num_latest =
             prov.get_block_by_number(BlockNumber::Latest).await.unwrap();
-        assert!(hash == block_by_num_latest.hash.unwrap());
-        let block_by_num = prov.get_block_by_number(1u64.into()).await.unwrap();
-        assert!(
-            block_by_num.transactions[0].hash
-                == block_by_num_latest.transactions[0].hash
-        );
+        let block_by_hash = prov
+            .get_block_by_hash(block_by_num_latest.hash.unwrap())
+            .await
+            .unwrap();
+        assert!(block_by_hash.hash == block_by_hash.hash);
     }
 
     // The test is ignored as the values used depend on the Geth instance used
     // each time you run the tests. And we can't assume that everyone will
     // have a Geth client synced with mainnet to have unified "test-vectors".
     #[ignore]
     #[tokio::test]
-    async fn test_trace_block_by_hash() {
-        let transport = Http::new(Url::parse("http://localhost:8545").unwrap());
+    async fn test_get_contract_code() {
+        let prov = get_provider();
+        let contract_address =
+            address!("0xd5f110b3e81de87f22fa8c5e668a5fc541c54e3d");
```

Comment:
Why is the address hardcoded? Won't it change?

Also not sure how we can get the addr of the deployed contract from the docker geth from here.

Any ideas @ed255 ?

Reply:
Technically I think this address shouldn't change because it's always created by the same account, and this account does a deterministic number of transactions.  The contract address will be determined by the deployer address and the deployer nonce.
It can change if later we add another transaction from the same account before the contract deployment.